### ISSUE TITLE
Can't use all-hies from a flake.nix

### ISSUE THREAD

**Issue by ju1m** [Infinisil/all-hies#73]
# Problem
Using `tools.hie = "unstable"` in a `flake.nix`'s `devShell`, one gets:
```
$ nix develop --show-trace
[…]
error: --- EvalError ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix
at: (90:39) in file: /nix/store/rkamf4cwqf8hbzr3sd7sq3x4qdzs6m3j-nixpkgs/pkgs/top-level/impure.nix                                                                                                                  
                                                                                                                                                                                                                    
    89|                 else (if args ? localSystem then {}                                                                                                                                                         
    90|                       else { system = builtins.currentSystem; }) // localSystem;                                                                                                                            
      |                                       ^                                                                                                                                                                     
    91| })                                                                                                                                                                                                          
                                                                                                                                                                                                                    
attribute 'currentSystem' missing
[…]
```

# Reproducing
`flake.nix`:
```
{
inputs.flake-utils.url = "github:numtide/flake-utils";
inputs.haskell-nix.url = "github:input-output-hk/haskell.nix";
inputs.all-hies.url = "github:infinisil/all-hies";
inputs.all-hies.flake = false;
outputs = inputs: let
  projectName = "myproject";
  in inputs.flake-utils.lib.eachDefaultSystem (system: let
    #pkgs = inputs.haskell-nix.legacyPackages.${system};
    pkgs = import inputs.haskell-nix.sources.nixpkgs (inputs.haskell-nix.nixpkgsArgs // {
      localSystem = { inherit system; };
      overlays = inputs.haskell-nix.nixpkgsArgs.overlays ++ [
        (import inputs.all-hies {}).overlay
      ];
    });
    compiler-nix-name = "ghc865";
    project = pkgs.haskell-nix.cabalProject {
      src = ./.;
      inherit compiler-nix-name;
    };
    in {
    devShell = project.shellFor {
      packages = hpkgs: [ hpkgs.${projectName} ];
      withHoogle = false;
      tools = {
        hie = "unstable";
      };
      exactDeps = true;
    };
    }
  );
}
```
`cabal.project`:
```
packages: .
```
# Note
This may be related to the use of `haskellNix.sources` and/or `haskellNix.nixpkgsArgs` here:
https://github.com/Infinisil/all-hies/blob/534ac517b386821b787d1edbd855b9966d0c0775/sources.nix#L7-L11
As those require special treatment in flake-mode like done here: https://github.com/input-output-hk/haskell.nix/blob/91f4e94ebc4274eac3587cee518b4ad9407e4f90/flake.nix#L15-L34
To avoid problem like this: https://github.com/input-output-hk/haskell.nix/issues/824

**Comment by ju1m** [comment]
I've just hit the exact same problem now using Nixpkgs' Haskell infrastructure. Moreover because current Nixpkgs is now with `glibc-2.31` which `all-hies` does not support yet, one must pin an older Nixpkgs with `glibc-2.30`, but can't use a `flake.nix` to do so because of this issue.
**Edit:** in the meantime a workaround could be to use `nix develop --impure` causing `builtins.currentSystem` to be available.

**Comment by infinisil** [comment]
As also mentioned in another issue, HIE isn't being maintained anymore, and I'll probably archive this repository to reflect that soon. The actively developed [HLS](https://github.com/haskell/haskell-language-server) has taken its place, and is also much easier to install since it's included in nixpkgs. I suggest giving that a try.

**Comment by ju1m** [comment]
@Infinisil Ack. Indeed, I was able to install `haskell-language-server` easily from the usual binaries cache for the current default `ghc884`, and AFAICS once running it's even much quicker than `hie` to report errors. Thanks nonetheless to you for all these months where `all-hies` helped me. 👍

As a vim+ALE users it looks like there is no specific `haskell-language-server` support yet, but I was able to reuse the current HIE support still there, just by changing `g:ale_haskell_hie_executable` from `hie-wrapper` to `haskell-language-server-wrapper`.
